### ISSUE TITLE
Cannot install soapyrtlsdr OS X 10.11.6

### ISSUE THREAD

**Issue by stuzog** [pothosware/homebrew-pothos#6]
I'm having trouble with CubicSDR on an iMac 7,1 OS X 10.11.6 following your instructions to uninstall and reinstall via Homebrew. I did this as CubicSDR was crashing on setup with a Nonelectric RTL-SDR dongle. When it comes to reinstalling soapystlsdr I get this:

```
bash$ brew install soapyrtlsdr
==> Installing soapyrtlsdr from pothosware/pothos
==> Downloading https://github.com/pothosware/SoapyRTLSDR/archive/soapy-rtlsdr-0.2.2.tar.gz
Already downloaded: /Users/Stuart/Library/Caches/Homebrew/soapyrtlsdr-0.2.2.tar.gz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr
==> make install
Last 15 lines from /Users/Stuart/Library/Logs/Homebrew/soapyrtlsdr/02.make:
[100%] Linking CXX shared module librtlsdrSupport.so
/usr/local/Cellar/cmake/3.7.1/bin/cmake -E cmake_link_script CMakeFiles/rtlsdrSupport.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -std=c++11 -Wc++11-extensions -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11 -bundle -Wl,-headerpad_max_install_names  -flat_namespace -o librtlsdrSupport.so CMakeFiles/rtlsdrSupport.dir/Registation.cpp.o CMakeFiles/rtlsdrSupport.dir/Settings.cpp.o CMakeFiles/rtlsdrSupport.dir/Streaming.cpp.o /usr/local/lib/librtlsdr.dylib /usr/local/lib/libSoapySDR.dylib 
Undefined symbols for architecture x86_64:
  "_rtlsdr_set_agc_mode", referenced from:
      SoapyRTLSDR::setGainMode(int, unsigned long, bool) in Settings.cpp.o
  "_rtlsdr_set_direct_sampling", referenced from:
      SoapyRTLSDR::writeSetting(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in Settings.cpp.o
  "_rtlsdr_set_offset_tuning", referenced from:
      SoapyRTLSDR::writeSetting(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in Settings.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [librtlsdrSupport.so] Error 1
make[1]: *** [CMakeFiles/rtlsdrSupport.dir/all] Error 2
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/pothosware/homebrew-pothos/issues

bash$ brew uninstall soapyrtlsdr
Error: No such keg: /usr/local/Cellar/soapyrtlsdr
bash$ 
```

**Comment by stuzog** [comment]
Nonelectric RTL-SDR dongle => NooElectric RTL-SDR dongle.

**Comment by guruofquality** [comment]
I have no idea. This is a pretty common installation combination w/ homebrew. Is this an unusual version librtlsdr, is it from homebrew, any special custom brew recipe for rtl?

Those functions like `rtlsdr_set_offset_tuning` are pretty standard to be found in librtlsdr. Example: https://github.com/librtlsdr/librtlsdr/blob/master/src/librtlsdr.c#L1549

**Comment by stuzog** [comment]
Hi Josh,

I installed rtlsdr using instructions from 
<https://www.hfunderground.com/wiki/RTL-SDR> by downloading the archive 
from <http://inst.eecs.berkeley.edu/~ee123/fa12/rtlsdr/rtlsdr_osx.zip> 
and copying over the relevant files.

Running rtl_test finds the RTL-SDR unit, but does not connect:

Found 1 device(s):
0: ezcap USB 2.0 DVB-T/DAB/FM dongle

Using device 0: ezcap USB 2.0 DVB-T/DAB/FM dongle
Failed to open rtlsdr device #0.

I'm wondering if the failure to connect is due to SIP in OS X 10.11?

Initially, I installed CubicSDR using the instructions from 
<http://cubicsdr.readthedocs.io/en/latest/building-cubicsdr/osx.html#build-osx>. 
CubicSDR would open but crash immediately after selecting the RTL-SDR 
unit. I then installed using your scripts but ran across the problem 
I've described.

Very frustrating -- I've tried every way of getting CubicSDR to work on 
my iMac7,1, but it eludes me.

Thaks for your work on your installation scripts,
Stuart

**Comment by stuzog** [comment]
Correction: I later installed rtlsdr using Homebrew, which I use instead 
of Macports, as shown here:

bash$ brew info librtlsdr
librtlsdr: stable 0.5.3 (bottled), HEAD
Use Realtek DVT-T dongles as a cheap SDR
https://sdr.osmocom.org/trac/wiki/rtl-sdr
/usr/local/Cellar/librtlsdr/0.5.3 (18 files, 317K) *
   Poured from bottle on 2016-12-11 at 09:27:03
From: 
https://github.com/Homebrew/homebrew-core/blob/master/Formula/librtlsdr.rb
==> Dependencies
Build: pkg-config ✔, cmake ✔
Required: libusb ✔
==> Options
--universal
     Build a universal binary
--HEAD
     Install HEAD version

So the version in use is 0.5.3. Should I use the HEAD version?

Stuart


Josh Blum wrote:

**Comment by stuzog** [comment]
Ah! That's better -- installing HEAD resulted in this:

🍺  /usr/local/Cellar/librtlsdr/HEAD-e3e6ee2: 19 files, 329.6K, built in 
19 seconds
bash$ sudo rtl_test
Password:
Found 1 device(s):
   0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Found Rafael Micro R820T tuner
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 
16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 
43.4 43.9 44.5 48.0 49.6
[R82XX] PLL not locked!
Sampling at 2048000 S/s.

Info: This tool will continuously read from the device, and report if
samples get lost. If you observe no further output, everything is fine.

Reading samples in async mode...
^CSignal caught, exiting!

User cancel, exiting...
Samples per million lost (minimum): 0
bash$

Stuart

Josh Blum wrote:

**Comment by stuzog** [comment]
Hi Josh,

Yes! It worked! I trashed the app version of CubicSDR compiled for OS X 
I had downloaded and installed it via Homebrew, then installed 
SoapyRTLSDL via the instructions at 
<https://github.com/cjcliffe/CubicSDR/wiki/Build-OSX>, and voila! 
CubicSDR works brilliantly.

Thanks so much for your clue about the version of librtlsdr, it started 
me off on the right track.

Happy holidays,
Stuart


Josh Blum wrote:
